iree/compiler/Dialect/LinalgExt/IR/test/invalid.mlir

```diff
@@ -380,3 +380,27 @@ func @scatter_original_rank_mismatch(
     } -> tensor<?x?xi64>
   return %0 : tensor<?x?xi64>
 }
+
+// -----
+
+func @reverse_diff_types(%arg0: tensor<3x5xi32>) -> tensor<3x6xi32> {
+  %init = linalg.init_tensor [3, 6] : tensor<3x6xi32>
+  // expected-error @+1 {{expected input/output types are identical}}
+  %0 = linalg_ext.reverse
```

Comment:
I definitely like this form better. This might mean that bufferization might create an extra buffer for these operations, i.e we need them to be `tied` somehow.

Reply:
I think we can not simply tie them together. Otherwise the result will get messed up. What is implemented in scalar form is

```cpp
for (int i =0; i < n; ++i) {
  out[i] = in[n - i - 1];
}
```

If we tie the input and output buffer, it will produce `1 2 3 2 1` if the input is `1 2 3 4 5`.

The previous definition fits more in tying buffers because it allows in-place update. However, it will break some assumptions in `TiledOpInteface`.